packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -45,33 +45,33 @@ exports.createFileNode = async (
     }
   }
 
-  // Stringify date objects.
-  return JSON.parse(
-    JSON.stringify({
-      // Don't actually make the File id the absolute path as otherwise
-      // people will use the id for that and ids shouldn't be treated as
-      // useful information.
-      id: createNodeId(pathToFile),
-      children: [],
-      parent: null,
-      internal,
-      sourceInstanceName: pluginOptions.name || `__PROGRAMMATIC__`,
-      absolutePath: slashedFile.absolutePath,
-      relativePath: slash(
-        path.relative(
-          pluginOptions.path || process.cwd(),
-          slashedFile.absolutePath
-        )
-      ),
-      extension: slashedFile.ext.slice(1).toLowerCase(),
-      size: stats.size,
-      prettySize: prettyBytes(stats.size),
-      modifiedTime: stats.mtime,
-      accessTime: stats.atime,
-      changeTime: stats.ctime,
-      birthTime: stats.birthtime,
-      ...slashedFile,
-      ...stats,
-    })
-  )
+  stats.atime = stats.atime.toJSON()
+  stats.mtime = stats.mtime.toJSON()
+  stats.ctime = stats.ctime.toJSON()
+  stats.birthtime = stats.birthtime.toJSON()
+
+  return {
+    // Don't actually make the File id the absolute path as otherwise
+    // people will use the id for that and ids shouldn't be treated as
+    // useful information.
+    id: createNodeId(pathToFile),
+    children: [],
+    parent: null,
+    internal,
+    sourceInstanceName: pluginOptions.name || `__PROGRAMMATIC__`,
+    relativePath: slash(
+      path.relative(
+        pluginOptions.path || process.cwd(),
+        slashedFile.absolutePath
+      )
+    ),
+    extension: slashedFile.ext.slice(1).toLowerCase(),
+    prettySize: prettyBytes(stats.size),
+    modifiedTime: stats.mtime,
+    accessTime: stats.atime,
+    changeTime: stats.ctime,
+    birthTime: stats.birthtime,
+    ...slashedFile,
+    ...stats,
+  }
```

Comment:
can't we do this? I know nits
```suggestion
  return {
    // Don't actually make the File id the absolute path as otherwise
    // people will use the id for that and ids shouldn't be treated as
    // useful information.
    id: createNodeId(pathToFile),
    children: [],
    parent: null,
    internal,
    sourceInstanceName: pluginOptions.name || `__PROGRAMMATIC__`,
    relativePath: slash(
      path.relative(
        pluginOptions.path || process.cwd(),
        slashedFile.absolutePath
      )
    ),
    extension: slashedFile.ext.slice(1).toLowerCase(),
    prettySize: prettyBytes(stats.size),
    ...stats,
    modifiedTime: stats.mtime.toJSON(),
    accessTime: stats.atime.toJSON(),
    changeTime: stats.ctime.toJSON(),
    birthTime: stats.birthtime.toJSON(),
    ...slashedFile,
  }
```

Reply:
Absolutely but it's backwards compat breaking and requires bumping a major.

Reply:
And in that case I'd opt to strip a few more properties because for the 64k site, those splats take 500ms each, some of these properties are 100ms (in aggregate). But at scale it's not likely to make the biggest dent and there might be a solid reason for those props to be exposed, so :shrug: 

In v3 we may want to reconsider the shape of this node and reduce it a bit to what is really needed, rather than dumping all the things in there like we do now.

Reply:
Current shape of this node:

```js
node = { id: '6b94aed0-6526-58f4-8a18-c35a21c0b6b9',                                                                                                                                                                                                                                                                               
  children: [],
  parent: null,
  internal:
   { contentDigest: 'bdb8b0c7092c676a650f2c5cb2c20807',
     type: 'File',
     mediaType: 'text/plain',
     description:
      'File "generated_articles/2020-10-18_minus-quae-sit-quibusdam-aliquam..txt"' 
  },
  sourceInstanceName: 'blog',
  absolutePath:
   'SNIP/benchmarks/gabe-fs-text/generated_articles/2020-10-18_minus-quae-sit-quibusdam-aliquam..txt',
  relativePath: '2020-10-18_minus-quae-sit-quibusdam-aliquam..txt',
  extension: 'txt',
  size: 506,
  prettySize: '506 B',
  modifiedTime: '2020-10-22T13:50:31.589Z',
  accessTime: '2020-10-22T13:50:31.589Z',
  changeTime: '2020-10-22T13:50:31.589Z',
  birthTime: '2020-10-22T13:50:31.589Z',
  root: '/',
  dir:
   'SNIP/benchmarks/gabe-fs-text/generated_articles',
  base: '2020-10-18_minus-quae-sit-quibusdam-aliquam..txt',
  ext: '.txt',
  name: '2020-10-18_minus-quae-sit-quibusdam-aliquam.',
  relativeDirectory: '',
  dev: 64769,
  mode: 33204,
  nlink: 1,
  uid: 1000,
  gid: 1000,
  rdev: 0,
  blksize: 4096,
  ino: 58858167,
  blocks: 8,
  atimeMs: 1603374631589.1082,
  mtimeMs: 1603374631589.1082,
  ctimeMs: 1603374631589.1082,
  birthtimeMs: 1603374631589.1082,
  atime: '2020-10-22T13:50:31.589Z',
  mtime: '2020-10-22T13:50:31.589Z',
  ctime: '2020-10-22T13:50:31.589Z',
  birthtime: '2020-10-22T13:50:31.589Z' 
}
```

Reply:
THan revert to the forEach I guess :joy:

Reply:
Do you want me to change the current version (where it explicitly updates the four Data keys on `stats`) back to a forEach loop? This is faster / more to the point and does not change the status quo.